pkg/hbone/README.md

```diff
@@ -0,0 +1,71 @@
+# HTTP Based Overlay Network (HBONE)
+
+HTTP Based Overlay Network (HBONE) is the protocol used by Istio for communication between workloads in the mesh.
+At a high level, the protocol consists of tunneling TCP connections over HTTP/2 CONNECT, over mTLS.
+
+## Specification
+
+TODO
```

Comment:
My doc on gateway improvements has several links, you can add them here ( RFCs, BeyondCorp, browser specs)

Reply:
+1 ... could you fill out this section?